### PR TITLE
feat(button): add ghost danger button

### LINKED ISSUE #19: Add "ghost danger" button variation
### Proposal

We have a "ghost danger" button in the Tearsheet, but application teams are going to need the "ghost danger" button in other contexts as well. Therefore, it would be best if we broke it out into a reusable component.

### PR DESCRIPTION
**Pull request by b3nk3**
## Affected issues

- Resolves #19 

## Proposed changes

- Adds 'ghost-danger' option to Button component
- Removes current ghost danger implementation from Tearsheet and uses the one implemented on Button

### CODE REVIEW

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -18,16 +18,23 @@ const { defaultProps, propTypes } = CarbonButton;
...
 
-const Button = ({ className, largeText, size, ...other }) => {
+const Button = ({ className, largeText, size, kind, ...other }) => {
   const isSize = value => size === value;
   const isLarge = isSize('large') || largeText || isSize('lg') || isSize('xlg');
 
+  const isKind = value => kind === value;
+  const isGhostDanger = isKind('ghost-danger');
+
+  console.log(!isGhostDanger ? kind : null);
```
Comment (jendowns) [inline src/components/Button/Button.js]: Could you remove this `console.log()`?

File: src/components/Button/Button.js [hunk 2]
```diff
@@ -18,16 +18,23 @@ const { defaultProps, propTypes } = CarbonButton;
...
+
   return (
     <CarbonButton
       className={classnames(namespace, className, {
         [`${namespace}--large`]: isLarge,
+        [`${namespace}--ghost-danger`]: isGhostDanger,
       })}
       size={!isLarge ? size : null}
+      kind={!isGhostDanger ? kind : 'ghost'}
```
Comment (jendowns) [inline src/components/Button/Button.js]: Just checking -- are you keeping the `kind='ghost'` in this case to get the Carbon ghost button classes?

Comment (b3nk3) [reply src/components/Button/Button.js]: Yes, as that is how the implementation worked in the Tearsheet component.

I am not too familiar with the carbon system, so there might be a more efficient and future proof way of doing this, if so, do let me know.

File: src/components/Button/Button.js (hunk 1 above)
Comment (b3nk3) [reply src/components/Button/Button.js]: Ooops, forgot about that. Soz.

File: src/components/Button/Button.js [hunk 3]
```diff
@@ -18,16 +18,22 @@ const { defaultProps, propTypes } = CarbonButton;
 
 // TODO: V3 - Remove deprecated props `largeText`.
 
-const Button = ({ className, largeText, size, ...other }) => {
+const Button = ({ className, largeText, size, kind, ...other }) => {
   const isSize = value => size === value;
   const isLarge = isSize('large') || largeText || isSize('lg') || isSize('xlg');
 
+  const isKind = value => kind === value;
+  const isGhostDanger = isKind('ghost-danger');
```
Comment (SimonFinney) [inline src/components/Button/Button.js]: - [ ] The `isKind` method is redundant here and can be removed if it's not being used elsewhere:

```suggestion

const isGhostDanger = kind === 'ghost-danger';
```

File: src/components/Button/Button.stories.js [hunk 4]
```diff
@@ -28,6 +28,7 @@ const kinds = {
   'Secondary button (secondary)': 'secondary',
   'Danger button (danger)': 'danger',
   'Ghost button (ghost)': 'ghost',
+  'Ghost Danger button (ghost-danger)': 'ghost-danger',
```
Comment (SimonFinney) [inline src/components/Button/Button.stories.js]: - [ ] I recommend using standard case type here for consistency:

```suggestion
'Ghost danger button (ghost-danger)': 'ghost-danger',
```

File: src/components/Button/Button.js (hunk 3 above)
Comment (b3nk3) [reply src/components/Button/Button.js]: Thanks for this, you are absolutely right.

File: src/components/Button/Button.js [hunk 5]
```diff
@@ -18,16 +18,23 @@ const { defaultProps, propTypes } = CarbonButton;
 
 // TODO: V3 - Remove deprecated props `largeText`.
 
-const Button = ({ className, largeText, size, ...other }) => {
+const Button = ({ className, largeText, size, kind, ...other }) => {
   const isSize = value => size === value;
   const isLarge = isSize('large') || largeText || isSize('lg') || isSize('xlg');
 
+  const isKind = value => kind === value;
```
Comment (SimonFinney) [inline src/components/Button/Button.js]: - [ ] Thanks! The final thing here is to remove this line now it's no longer needed:

```suggestion
```

**Pull request merged**